colossus-metrics/src/main/scala/colossus/metrics/MetricClock.scala

```diff
@@ -39,29 +37,24 @@ class IntervalAggregator(interval: FiniteDuration,
       reporters.foreach { reporter =>
         reporter ! ReportMetrics(build)
       }
-    }
 
-    case RegisterCollection(collection) => {
+    case RegisterCollection(collection) =>
       collections = collections + collection
-    }
 
-    case RegisterReporter(ref) => {
+    case RegisterReporter(ref) =>
       reporters = reporters + ref
       context.watch(ref)
-    }
 
-    case Terminated(child) => {
+    case Terminated(child) =>
       if (reporters.contains(child)) {
         debug(s"oh no!  We lost a MetricReporter $child. Removing from registered reporters.")
-        reporters.remove(child)
+        reporters = reporters - child
```

Comment:
glad you found this bug. I was fixing this one in a similar branch addressing these issue :)